src/saltext/vmware/states/vmc_dhcp_profiles.py

```diff
@@ -0,0 +1,420 @@
+"""
+VMC DHCP Profiles state module
+
+Add new DHCP profile and delete existing DHCP profile from an SDDC.
+
+Example usage :
+
+.. code-block:: yaml
+
+    ensure_dhcp_profile:
+      vmc_dhcp_profiles.present:
+        - hostname: sample-nsx.vmwarevmc.com
+        - refresh_key: 7jPSGSZpCa8e5Ouks4UY5cZyOtynAhF
+        - authorization_host: console-stg.cloud.vmware.com
+        - org_id: 10e1092f-51d0-473a-80f8-137652c39fd0
+        - sddc_id: b43da080-2626-f64c-88e8-7f31d9d2c306
+        - type: server
+        - dhcp_profile_id: dhcp-test
+        - verify_ssl: False
+        - cert: /path/to/client/certificate
+
+.. warning::
+
+    It is recommended to pass the VMC authentication details using Pillars rather than specifying as plain text in SLS
+    files.
+
+
+"""
+import logging
+
+from saltext.vmware.utils import vmc_constants
+from saltext.vmware.utils import vmc_state
+
+log = logging.getLogger(__name__)
+DHCP_PROFILE_NOT_FOUND_ERROR = "could not be found"
+
+
+try:
+    from saltext.vmware.modules import vmc_dhcp_profiles
+
+    HAS_DHCP_PROFILES = True
+except ImportError:
+    HAS_DHCP_PROFILES = False
+
+
+def __virtual__():
+    if not HAS_DHCP_PROFILES:
+        return False, "'vmc_dhcp_profiles' binary not found on system"
```

Comment:
**suggestion** This message is incorrect - in fact, looking at `modules/vmc_dhcp_profiles.py` the import should never fail. Either `__virtual__` and the associated ImportError block should be completely removed, or virtual should just return the string.

Reply:
we followed this approach in states/vmc_security_rules.py based on review comments. So we applied it across all the state files.

Reply:
https://github.com/saltstack/salt-ext-modules-vmware/pull/84#pullrequestreview-770849368
Although the approach of implementation is similar through state files, the request params and logs are very particular and as we already wrote all the unit and integration tests across all the modules based on this approach. we wouldn't be able to address this change for now as it will take lots of efforts and time.